test/it/sms-template.js

```diff
@@ -0,0 +1,158 @@
+      template.delete();
+    });
+
+    it('should return correct model', async () => {
+      template = await client.createSmsTemplate(generalFakeTemplateObj);
+      expect(template).to.be.instanceOf(models.SmsTemplate);
+    });
+
+    it('should return correct data with id assigned', async () => {
+      template = await client.createSmsTemplate(generalFakeTemplateObj);
+      expect(template).to.have.property('id');
+      expect(template.name).to.equal(generalFakeTemplateObj.name);
+    });
+  });
+
+  describe('Delete template', () => {
+    let template;
+    beforeEach(async () => {
+      template = await client.createSmsTemplate(generalFakeTemplateObj);
+    });
+
+    afterEach(async () => {
+      await template.delete();
+    });
+
+    it('should not get template after deletion', async () => {
+      await client.deleteSmsTemplate(template.id);
+      try {
+        await client.getSmsTemplate(template.id);
+      } catch (e) {
+        expect(e.status).to.equal(404);
+      }
+    });
+  });
+
+  describe('Get template', () => {
+    let template;
+    beforeEach(async () => {
+      template = await client.createSmsTemplate(generalFakeTemplateObj);
+    });
+
+    afterEach(async () => {
+      await template.delete();
+    });
+
+    it('should get SmsTemplate by id', async () => {
+      const templateFromGet = await client.getSmsTemplate(template.id);
+      expect(templateFromGet).to.be.instanceOf(models.SmsTemplate);
+      expect(templateFromGet.name).to.equal(generalFakeTemplateObj.name);
+    });
+  });
+
+  describe('Partial Update template', () => {
+    let template;
+    beforeEach(async () => {
+      template = await client.createSmsTemplate(generalFakeTemplateObj);
+    });
+
+    afterEach(async () => {
+      await template.delete();
+    });
+
+    it('should update template name property', async () => {
+      const updatedTemplate = await client.partialUpdateSmsTemplate(template.id, { name: 'fake updated name' });
+      expect(updatedTemplate).to.be.instanceOf(models.SmsTemplate);
+      expect(updatedTemplate.id).to.equal(template.id);
+      expect(updatedTemplate.name).to.equal('fake updated name');
+      expect(updatedTemplate.template).to.equal(generalFakeTemplateObj.template);
+    });
+  });
+
+  describe('Update template', () => {
+    let template;
+    beforeEach(async () => {
+      template = await client.createSmsTemplate(generalFakeTemplateObj);
+    });
+
+    afterEach(async () => {
+      await template.delete();
+    });
+
+    it('should update all properties in template', async () => {
+      const updatedTemplate = await client.updateSmsTemplate(template.id, {
+        name: 'fake updated name',
+        type: 'fake updated type',
+        template: 'Your fake updated verification code is ${code}.'
+      });
+      expect(updatedTemplate.name).to.equal('fake updated name');
+      expect(updatedTemplate.translations).to.be.undefined;
+
+      // Clean
+      updatedTemplate.delete();
```

Comment:
Is this redundant redundant?

Reply:
I should add more comment here. Basically if it updates with changed template type, a new resource will be created in backend. This clean step is used to make sure that new created resource can also be cleaned up.

Comment will be added before merge.